packages/react-devtools-shared/src/utils.js

```diff
@@ -598,37 +599,44 @@ export function getDataType(data: Object): DataType {
         return 'number';
       }
     case 'object':
-      if (isArray(data)) {
-        return 'array';
-      } else if (ArrayBuffer.isView(data)) {
-        return hasOwnProperty.call(data.constructor, 'BYTES_PER_ELEMENT')
-          ? 'typed_array'
-          : 'data_view';
-      } else if (data.constructor && data.constructor.name === 'ArrayBuffer') {
-        // HACK This ArrayBuffer check is gross; is there a better way?
-        // We could try to create a new DataView with the value.
-        // If it doesn't error, we know it's an ArrayBuffer,
-        // but this seems kind of awkward and expensive.
-        return 'array_buffer';
-      } else if (typeof data[Symbol.iterator] === 'function') {
-        const iterator = data[Symbol.iterator]();
-        if (!iterator) {
-          // Proxies might break assumptoins about iterators.
-          // See github.com/facebook/react/issues/21654
+      try {
+        if (Array.isArray(data)) {
```

Comment:
I don’t think this can throw. Or can it? Can we scope down even more to specific places that can throw? Several try-catch is fine.

Reply:
@gaearon I don't think I can `try-catch` wrap something in a `if (/* here */)` expression unless I create a helper function. Should I do that?

Reply:
```js
let condition = false
try {
  condition = ...
} catch (err) {
 ...
}
if (condition) {

}
```

?

Reply:
it will certanly look odd, but I can do it like this